services/cloudapi/cloudapiserver/server.go

```diff
@@ -121,6 +179,80 @@ func (ss *Service) UpdateVmState(ctx context.Context, datacenterID, serverID, ne
 	return nil
 }
 
+func (ss *Service) UpdateBootDevice(ctx context.Context, datacenterID, serverID, newBootDeviceID string) error {
+	oldBootDeviceID, oldBdType, err := ss.GetCurrentBootDeviceID(ctx, datacenterID, serverID)
+	if err != nil {
+		return err
+	}
+	if strings.EqualFold(oldBootDeviceID, newBootDeviceID) {
+		return nil
+	}
+
+	newBdType := BootDeviceTypeCDROM
+	_, apiResponse, err := ss.Client.ImagesApi.ImagesFindById(ctx, newBootDeviceID).Execute()
+	if err != nil {
+		if !(apiResponse != nil && apiResponse.StatusCode == 404) {
+			return err
+		}
+		log.Printf("[DEBUG] no bootable image found with id : %s\n", newBootDeviceID)
+		newBdType = BootDeviceTypeVolume
+	}
+
+	switch oldBdType {
+	case BootDeviceTypeCDROM:
+		if strings.EqualFold(newBdType, BootDeviceTypeVolume) {
+			sp := ionoscloud.ServerProperties{BootVolume: ionoscloud.NewResourceReference(newBootDeviceID)}
+			if _, _, err = ss.Update(ctx, datacenterID, serverID, sp); err != nil {
+				return err
+			}
+			// detach the cdrom
+			apiResponse, err = ss.Client.ServersApi.DatacentersServersCdromsDelete(ctx, datacenterID, serverID, oldBootDeviceID).Execute()
+			if err != nil {
+				return err
+			}
+			if errState := cloudapi.WaitForStateChange(ctx, ss.Meta, ss.D, apiResponse, schema.TimeoutUpdate); errState != nil {
+				return errState
+			}
+			return nil
```

Comment:
add a log here where we mention what/who was detached

Reply:
refactored the logic a bit and added some extra logging - [90b61bb](https://github.com/ionos-cloud/terraform-provider-ionoscloud/pull/507/commits/90b61bb22669e6c0866d253d4535805c673b7c73)